### PR TITLE
Added agents detect command issue #61

### PR DESCRIPTION
**Pull request by donabp**
Creating a pull request and submitting it for review for issue https://github.com/corgibytes/freshli-cli/issues/61 .

### CODE REVIEW

File: Corgibytes.Freshli.Cli/Commands/AgentsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                     filesResults = Directory.GetFiles(path);
+                 }
+                foreach(string file in filesResults){
+                    if (Path.GetFileName(file).Contains("freshli-agent-")) {
+                        agents.Add(file);
+                    }
+                }
+            }
+            
+            agents = agents.Distinct().ToList();
```
Comment (mscottford) [inline Corgibytes.Freshli.Cli/Commands/AgentsCommand.cs:59]: This logic should be moved into a class named something like `AgentDetector` or something to that effect. That way we can more easily write unit tests against this functionality.

File: Corgibytes.Freshli.Cli/Commands/AgentsCommand.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        private int Run(IHost host, InvocationContext context, AgentsDetectCommandOptions options)
+        {
+            string [] paths =System.Environment.GetEnvironmentVariable("PATH").Split(":");
+            
+            List<string> agents = new List<string>();
+            string[] filesResults = null;
+
+            foreach (string path in paths) {
+                
+                 if(path.Contains("~/")){
```
Comment (mscottford) [inline Corgibytes.Freshli.Cli/Commands/AgentsCommand.cs:44]: It's going to be best to store `~/` in a constant. That's something that could go in the `Environment` class. Perhaps calling it `Environment.HomeDirectoryPlaceholder`.

File: Corgibytes.Freshli.Cli/Commands/AgentsCommand.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        public AgentsDetectCommand() : base("detect",
+            "Outputs the detected language name and the path to the language agent binary in a tabular format")
+        {
+            Handler = CommandHandler.Create<IHost, InvocationContext, AgentsDetectCommandOptions>(Run);
+        }
+
+        private int Run(IHost host, InvocationContext context, AgentsDetectCommandOptions options)
+        {
+            string [] paths =System.Environment.GetEnvironmentVariable("PATH").Split(":");
```
Comment (mscottford) [inline Corgibytes.Freshli.Cli/Commands/AgentsCommand.cs:37]: Instead of using `:` here, it's best to use `Path.PathSeparator`.

**Review by mscottford — CHANGES_REQUESTED** [review]
Yay! Your first pull request on the project! This is great!

I've made a _ton_ of comments here. Let's work on them together. Please don't read these and think that you've done a bad job. 

It's clear that the functionality works. You understood the problem and you implemented it. That's great! 

I just want us to make some adjustments to how the implementation is structured.

File: Corgibytes.Freshli.Cli/Commands/AgentsCommand.cs (hunk 2 above)
Comment (donabp) [reply Corgibytes.Freshli.Cli/Commands/AgentsCommand.cs:44]: are you saying I should create a new class called Environment and have Environment.HomeDirectoryPlaceholder as a field, or should I find an object with this field?

File: Corgibytes.Freshli.Cli/Commands/AgentsCommand.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                foreach(string file in filesResults){
+                    if (Path.GetFileName(file).Contains("freshli-agent-")) {
+                        agents.Add(file);
+                    }
+                }
+            }
+            
+            agents = agents.Distinct().ToList();
+            foreach(string agent in agents){
+                Console.WriteLine("The following agent was detected: " + agent );
```
[the remark this answers is not in the document]
Comment (donabp) [reply Corgibytes.Freshli.Cli/Commands/AgentsCommand.cs:61]: did you want me just to add whitespace here or is there a method that I should be calling? im going to search through the codebase for some other examples of such output

**Review by edwinkortman — APPROVED** [review]
Approved as an ensemble with @donabp, @mscottford, and @CodeMouse92 .

**Pull request merged**